src/perftest_resources.c

```diff
@@ -3416,8 +3416,15 @@ int run_iter_bw(struct pingpong_context *ctx,struct perftest_parameters *user_pa
 							goto cleaning;
 						}
 
-						ctx->ccnt[wc_id] += user_param->cq_mod;
-						totccnt += user_param->cq_mod;
+						if (ctx->ccnt[wc_id] + user_param->cq_mod <= user_param->iters) {
```

Comment:
Couldn't this if else condition cause performance issues / degradations ?

Reply:
i didn't notice, but you are right, someone could see a little overhead.
Do you have any suggestions?
i think i see a few options to make it a little better:

1. add starting condition, that cq_mod must be a divider of iters
2. do something like:
```
int fill = min(cq_mod, user_param->iters - ctx->ccnt[wc_id]);
ctx->ccnt[wc_id] += fill;
totccnt += fill;
```
but i suppose the if statement would be just hidden here...
3. instead of doing this for every iteration in this loop, do the check after the loop and alter it if needed. i'd add something like this into line 3442:
```
for(j=0;j<num_of_qps;j++) {
  if (ctx->ccnt[j] > user_param->iters) {
    totccnt -= user_param->iters - ctx->ccnt[j];
    ctx->ccnt[j] =user_param->iters;
  }
}

``` 
even though there's an additional loop, some time should be saved for several reasons: in noPeak==OFF scenario it would be done after completion time is gathered;
in situations when more than 1 completion is polled we'd get from 2 to tx_depth times less checking.

alternatively, we could avoid the for loop and put it in line 3319, but i'm not sure we wouldn't still prematurely escape the while loop at line 3314

Any other thoughts?